doc/source/examples/sharp-immersed-boundary/3d-rbf-static-mixer/3d-rbf-static-mixer.rst

```diff
@@ -0,0 +1,297 @@
+#. Operation ``15`` forms the casing, and operation ``16`` joins the casing and the helix. The final operation is the one considered as definitive.
+
+.. code-block:: text
+
+    shapes
+    0; rbf             ; RBF_helix.output;   -76.201:-20.0098:+15.6051;    0:+1.57079632679:0
+    1; hyper rectangle ;         75:25:25;                       0:0:0;                 0:0:0
+    2; cylinder        ;         15:10000;                       0:0:0;    0:+1.57079632679:0
+    operations
+    15; difference     ; 2:1
+    16; union          ; 0:15
+
+---------------
+Parameter File
+---------------
+
+Simulation Control
+~~~~~~~~~~~~~~~~~~
+
+Although we are interested in the steady-state solution of the flow, we use ``bdf1`` time integration. The required time to reach steady state in our case is low, but resolving it with low time steps enables the non-linear solver to converge as complex flow patterns are difficult to capture otherwise.
+
+.. code-block:: text
+
+    subsection simulation control
+      set method      = bdf1
+      set time end    = 40e-4
+      set time step   = 1e-4
+      set output path = ./output/
+      set output name = output
+    end
+
+Physical Properties
+~~~~~~~~~~~~~~~~~~~
+
+We assume that the used fluid is water, and that the static mixer is of the meter length scale (150 centimeters). The time unit is seconds and the length unit is centimeter.
+
+.. code-block:: text
+
+    subsection physical properties
+      subsection fluid 0
+        set kinematic viscosity = 0.01
+      end
+    end
+
+
+Mesh and Mesh Adaptation
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The mesh is a simple hyper rectangle, large enough to encompass the mixer with its casing and long enough to establish the flow profile upstream and downstream.
+
+
+.. code-block:: text
+
+    subsection mesh
+      set type      = dealii
+      set grid type = subdivided_hyper_rectangle
+
+      # Grid to use when solving the flow in the long version of the mixer
+      set grid arguments = 6,1,1: -150,-25,-25: 150,25,25: true
+
+      set initial refinement = 3
+    end
+
+Mesh adaptation ``type`` is set to ``kelly``, to allow adaptive refinement at the solid surface. This is necessary for simulations of this type because of the prohibitive mesh size required when only uniform refinement is used. Setting ``max refinement level = 5`` allows for two levels of adaptive refinement from the uniform ``initial refinement = 3`` defined in the ``mesh`` section. The ``frequency = 10000`` to ensure that no refinement occurs between time steps, as they are not necessary here.
+
+.. code-block:: text
+
+    subsection mesh adaptation
+      set type                 = kelly
+      set fraction type        = number
+      set max number elements  = 2000000
+      set max refinement level = 5
+      set min refinement level = 0
+      set frequency            = 100000
+    end
```

Comment:
Could you instead use `set frequency = 0` and `set initial refinement steps = 2`?

Reply:
I'm not sure I understand how this mesh refinement works. It is performed only one time at the beginning with the velocity as the variable. But is the initial velocity=(1,0,0) every where except in the solid and it transitions from (1,0,0) to (0,0,0) at the surface? 

I would maybe add a figure of this mesh (if we can see something :) ).

Reply:
I changed the code and switched to `frequency=0`